frontend/src/components/project/ProjectCommentsContent.js

```diff
@@ -26,7 +29,21 @@ export default function CommentsContent({ user, project, token, setCurComments }
   const comments = project.comments;
 
   const handleRemoveComment = (c) => {
-    setCurComments([...project.comments.filter((pc) => pc.id !== c.id)]);
+    // removing a top comment
+    if (c.parent_comment_id === null) {
+      setCurComments([...project.comments.filter((pc) => pc.id !== c.id)]);
+
+      // remove a comment that has a parent comment
+    } else {
+      const parentCommentIndex = project.comments.findIndex(
+        (comment) => comment.id === c.parent_comment_id
+      );
+      const filterOutReplies = [
+        ...project.comments[parentCommentIndex].replies.filter((pc) => pc.id !== c.id),
+      ];
+      project.comments[parentCommentIndex].replies = filterOutReplies;
+      setCurComments([...project.comments]);
```

Comment:
However with my solution from IdeasBoard this line won't work anymore since the new value of `project` is only applied after the function finishes running.